src/components/calcite-checkbox/calcite-checkbox.tsx

```diff
@@ -73,6 +81,10 @@ export class CalciteCheckbox {
 
   /** True if the checkbox is disabled */
   @Prop({ reflect: true }) disabled?: boolean = false;
+  @Watch("disabled")
+  disabledChanged(disabled: boolean) {
+    this.input.disabled = disabled;
+  }
```

Comment:
These added watchers (for disabled and name) are unrelated to the original issue but thought I'd include them for completeness like radio-button does.